packages/kbn-es-query/src/filters/build_filters/get_filter_params.ts

```diff
@@ -7,24 +7,30 @@
  */
 
 import type { PhrasesFilter } from './phrases_filter';
-import type { PhraseFilter } from './phrase_filter';
-import type { RangeFilter } from './range_filter';
+import { isPhraseFilter } from './phrase_filter';
+import { isMetaRangeFilter } from './range_filter';
 import { Filter, FILTERS } from './types';
 
 /**
  * @internal used only by the filter bar to create filter pills.
  */
-export function getFilterParams(filter: Filter) {
+export function getFilterParams(filter: Filter): Filter['meta']['params'] {
   switch (filter.meta.type) {
     case FILTERS.PHRASE:
-      return (filter as PhraseFilter).meta.params.query;
+      if (isPhraseFilter(filter)) {
+        return filter.meta.params?.query;
+      }
     case FILTERS.PHRASES:
       return (filter as PhrasesFilter).meta.params;
     case FILTERS.RANGE:
-      const { gte, gt, lte, lt } = (filter as RangeFilter).meta.params;
-      return {
-        from: gte ?? gt,
-        to: lt ?? lte,
-      };
+      if (isMetaRangeFilter(filter) && filter.meta.params) {
+        const { gte, gt, lte, lt } = filter.meta.params;
+        return {
+          from: gte ?? gt,
+          to: lt ?? lte,
+        };
+      }
+    default:
+      return filter.meta.params;
```

Comment:
if type guards have to be added inside each case, then can we drop completely the `switch` structure?

```ts
export function getFilterParams(filter: Filter): Filter['meta']['params'] {
      if (isPhraseFilter(filter)) {
        return filter.meta.params?.query;
      }
      if(isPhrasesFilter(filter)) {
        return filter.meta.params;
      }
      if (isMetaRangeFilter(filter) && filter.meta.params) {
        const { gte, gt, lte, lt } = filter.meta.params;
        return {
          from: gte ?? gt,
          to: lt ?? lte,
        };
      }
      return filter.meta.params;
```

Same fucntion length as before, better type safety and still clear to read. Wdyt?

Reply:
@dej611 ya agree completely, will update.